core/src/lib.rs

```diff
@@ -80,6 +82,142 @@ pub struct UpdateComparingParams<CRUD: Crudable> {
 pub type MokaFutureCrudableMap<CRUD> =
     moka::future::Cache<<CRUD as Crudable>::Pkey, Arc<arc_swap::ArcSwap<CRUD>>>;
 
+#[derive(Clone)]
+pub struct CrudableMapWithMoka<CRUD: Crudable> {
+    redis: redis_cache::RedisCrudableMap<CRUD>,
+    moka: Option<MokaFutureCrudableMap<CRUD>>,
+    use_moka: bool,
+}
+
+impl<CRUD> CrudableMapWithMoka<CRUD>
+where
+    CRUD: Crudable + serde::Serialize + for<'de> serde::Deserialize<'de>,
+    CRUD::Pkey: redis::ToRedisArgs + redis::FromRedisValue + std::fmt::Debug,
+    CRUD::MonoField: serde::Serialize + for<'de> serde::Deserialize<'de>,
+{
+    pub fn new(
+        redis_config: redis_cache::RedisConfig,
+        moka: Option<MokaFutureCrudableMap<CRUD>>,
+    ) -> Result<Self, redis::RedisError> {
+        let redis = redis_cache::RedisCrudableMap::new(redis_config)?;
+        let use_moka = moka.is_some();
+
+        Ok(Self {
+            redis,
+            moka,
+            use_moka,
+        })
+    }
+
+    pub fn redis_only(redis_config: redis_cache::RedisConfig) -> Result<Self, redis::RedisError> {
+        Ok(Self {
+            redis: redis_cache::RedisCrudableMap::new(redis_config)?,
+            moka: None,
+            use_moka: false,
+        })
+    }
+
+    pub fn with_moka(
+        redis_config: redis_cache::RedisConfig,
+        moka: MokaFutureCrudableMap<CRUD>,
+    ) -> Result<Self, redis::RedisError> {
+        Ok(Self {
+            redis: redis_cache::RedisCrudableMap::new(redis_config)?,
+            moka: Some(moka),
+            use_moka: true,
+        })
+    }
+}
+
+#[async_trait]
+impl<CRUD> CrudableMap<CRUD> for CrudableMapWithMoka<CRUD>
+where
+    CRUD: Crudable + serde::Serialize + for<'de> serde::Deserialize<'de>,
+    CRUD::Pkey: redis::ToRedisArgs + redis::FromRedisValue + std::fmt::Debug,
+    CRUD::MonoField: serde::Serialize + for<'de> serde::Deserialize<'de>,
+{
+    async fn insert(&self, items: Vec<CRUD>) -> Vec<Arc<CRUD>> {
+        if self.use_moka {
+            if let Some(moka) = &self.moka {
+                let results = CrudableMap::insert(moka, items.clone()).await;
+
+                let redis_clone = self.redis.clone();
+                let items_clone = items;
+                tokio::spawn(async move {
+                    let _ = CrudableMap::insert(&redis_clone, items_clone).await;
+                });
+
+                return results;
+            }
+        }
+
+        CrudableMap::insert(&self.redis, items).await
```

Comment:
From above, you moved the logic of dispatching vs waiting for redis to the redis cache

```suggestion
    async fn insert(&self, items: Vec<CRUD>) -> Vec<Arc<CRUD>> {
            if let Some(moka) = &self.moka {
                CrudableMap::insert(moka, items.clone()).await;
            }

        self.redis.insert(items).await
```